### PR TITLE
refactor(Breadcrumbs.js): atom Breadcrumb component cleanup

### PR DESCRIPTION
**Pull request by ASchmidt1024**
little things and small css improvements

### CODE REVIEW

File: src/es/components/atoms/breadcrumb/Breadcrumb.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { Shadow } from '../../prototypes/Shadow.js'
  */
 export default class Breadcrumb extends Shadow() {
   constructor (options = {}, ...args) {
-    super(Object.assign(options, { importMetaUrl: import.meta.url }), ...args)
+    super({ ...options, importMetaUrl: import.meta.url }, ...args)
```
Comment (ASchmidt1024) [inline src/es/components/atoms/breadcrumb/Breadcrumb.js:14]: The options object is first spread using the ... operator before assigning import.meta.url. 
This approach avoids modifying the original options object.

File: src/es/components/atoms/breadcrumb/Breadcrumb.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default class Breadcrumb extends Shadow() {
    * @return {boolean}
    */
   shouldRenderCSS () {
-    return !this.root.querySelector(`:host > style[_css], ${this.tagName} > style[_css]`)
+    return !this.querySelector(`:host > style[_css], ${this.tagName} > style[_css]`)
```
Comment (ASchmidt1024) [inline src/es/components/atoms/breadcrumb/Breadcrumb.js:27]: the `root` property was replaced by `this` to simplify the selector and improve performance

Comment (Weedshaker) [reply src/es/components/atoms/breadcrumb/Breadcrumb.js:27]: the selector on this only will not work into the shadowDom. There you must use this.root

File: src/es/components/atoms/breadcrumb/Breadcrumb.js [hunk 3]
```diff
@@ -33,10 +33,15 @@ export default class Breadcrumb extends Shadow() {
...
+    const icon = this.getAttribute('icon')
+    const color = 'var(--color, pink)'
+    const textDecoration = 'var(--text-decoration, none)'
+
     this.css = /* css */`
       :host {
-        margin: var(--margin, 0) !important;
-        width: var(--width, 0) !important;
+        margin: var(--margin, 0); /* removed "!important" and added a default value */
+        width: var(--width, 0); /* removed "!important" and added a default value */
```
Comment (Weedshaker) [inline src/es/components/atoms/breadcrumb/Breadcrumb.js:44]: this margin and width with fallback 0 is questionable and possible not needed. needs to be testet... if needed for some reason leave !important to overseed the width and margin being set by o-body to all children within main. Also !important is not as ugly within a webcomponent as it used to be with a global only stylesheet.

File: src/es/components/atoms/breadcrumb/Breadcrumb.js [hunk 4]
```diff
@@ -50,39 +55,39 @@ export default class Breadcrumb extends Shadow() {
...
         background-position: center;
         background-repeat: no-repeat;
         background-size: 1em;
-        ${this.getAttribute('icon') !== 'false' ? 'color: transparent;' : ''}
+        ${icon !== 'false' ? 'color: transparent;' : ''}
         min-width: 1em;
       }
       @media only screen and (max-width: _max-width_) {
         :host {
-          margin: var(--margin-mobile, var(--margin, 0)) !important;
```
Comment (Weedshaker) [inline src/es/components/atoms/breadcrumb/Breadcrumb.js:84]: this margin and width with fallback 0 is questionable and possible not needed. needs to be testet... if needed for some reason leave !important to overseed the width and margin being set by o-body to all children within main. Also !important is not as ugly within a webcomponent as it used to be with a global only stylesheet.

**Review by Weedshaker — CHANGES_REQUESTED** [review]
this.root vs this

and

margin/width

please test it. THX

File: src/es/components/atoms/breadcrumb/Breadcrumb.js (hunk 2 above)
Comment (ASchmidt1024) [reply src/es/components/atoms/breadcrumb/Breadcrumb.js]: 👀 Okay. Rolled back.

File: src/es/components/atoms/breadcrumb/Breadcrumb.js (hunk 3 above)
Comment (ASchmidt1024) [reply src/es/components/atoms/breadcrumb/Breadcrumb.js:44]: 🤔 You mean in the case o-body is set for breadcrumb AND other components together? Then I will roll back it again.

**Review by Weedshaker — APPROVED** [review]

**Pull request merged**